includes/alerts.inc.php

```diff
@@ -879,3 +884,28 @@ function ExtTransports($obj)
         echo '; ';
     }
 }//end ExtTransports()
+
+/**
+ * Check if a device's parent is down
+ * Returns true if parent is down
+ * @param int $device Device-ID
+ * @return bool
+ */
+function IsParentDown($device)
+{
```

Comment:
This whole function could be one sql query, but I'm fine with it like this.

Reply:
Figured out the SQL, also, we need to take disable and ignore into account here I think.
```sql
SELECT 1 from devices as d
  LEFT JOIN devices_attribs as a USING (device_id)
  LEFT JOIN device_relationships as r ON d.device_id=r.parent_device_id
WHERE d.status = 0
  AND d.ignore=0
  AND d.disabled=0
  AND r.child_device_id=?
  AND (d.status_reason='icmp' OR (a.attrib_type='override_icmp_disable' AND a.attrib_value=true))
```

Reply:
With this query we still need to check how many parents current device has, and compare the result with count(1) or something. For example, if our child device has 3 parents and only 2 of them are down, we still need to run issueAlert(), don't you think?

Reply:
@aldemera Yeah, we could do counts and compare them.   I'll think about it more.

Reply:
Yeah, count parents and count down parents.

2 queries is much better than 1+n

Reply:
You can even skip the second query if parents == 0 ;)